llvm/unittests/SandboxIR/PassTest.cpp

```diff
@@ -133,3 +133,33 @@ define void @foo() {
   EXPECT_EQ(Buff, "test-fpm(test-pass1,test-pass2)");
 #endif // NDEBUG
 }
+
+TEST_F(PassTest, PassRegistry) {
+  class TestPass1 final : public FunctionPass {
+  public:
+    TestPass1() : FunctionPass("test-pass1") {}
+    bool runOnFunction(Function &F) final { return false; }
+  };
+  class TestPass2 final : public FunctionPass {
+  public:
+    TestPass2() : FunctionPass("test-pass2") {}
+    bool runOnFunction(Function &F) final { return false; }
+  };
+
+  PassRegistry Registry;
+  auto *TP1 = Registry.registerPass(std::make_unique<TestPass1>());
+  auto *TP2 = Registry.registerPass(std::make_unique<TestPass2>());
+
+  // Check getPassByName().
+  EXPECT_EQ(Registry.getPassByName("test-pass1"), TP1);
+  EXPECT_EQ(Registry.getPassByName("test-pass2"), TP2);
+
+  FunctionPassManager FPM("test-fpm");
```

Comment:
copy pasta?

Reply:
Yeah we don't really need a pass manager in this test.